### ISSUE TITLE
refactor: migrate batch download to AMQP

### ISSUE THREAD

**Issue by ClemDoum** [ICIJ/datashare#1081]
Depends on #1079

Let's start with batch download that only uses the databus (no DB).

* [x]  remove `save(Task)` that was made for progress to use event bus
* [x]  replace `Queue<BatchDownload>` by `TaskSupplier`  (in TaskManagerRedis, BatchDownloadLoop)
* [x]  make task manager memory display progress
* [x]  make task manager redis display progress
* [x]  make batchDownloadLoop run in background as batchSeachLoop in local mode
* [x]  refactor `startTask(Callable<V> task, Map<String, Object> properties)` to  `startTask(Class<T> taskName, Map<String, Object> properties)` and  make changes to task managers (memory, redis) to make this refactor work
* [x]  implement `startTask(Class<T> taskName, Map<String, Object> properties)` for AMQPTaskManager 
* [x]  implement `TaskModifier`, `TaskSupplier` for AMQP
* [x] allow the CLI/modes to instantiate AMQP `TaskManager`/`TaskModifier`/`TaskSupplier`
* [x] in local mode instantiate Qpid broker
* [x] test manually that batch download is working with AMQP and no-regression for Memory/Redis 

The goal is to uncouple the task scheduling (that is made from the "client" and could be remote) from task executing that will be made by "generic" task runners somewhere in the network.

The idea is to push "taskview" that is mainly a task "name" and task properties map. In  a first move, task name could be used to instantiate by introspection real executable tasks, providing task properties to the constructor.

These tasks are sent over the wire when scheduling. For the update of the tasks (progress, result, errors) it will be sent by events in another AMQP channel. So we have to think carefully how it could work for the local/memory case.

The smallest move found for now is to split TaskManager that was executing tasks into : 

```java
@FunctionalInterface
public interface TaskModifier {
    void progress(String taskId, double rate);
}
```

whose responsibility is to allow task runners to update tasks (maybe it could be split into 3 functional interfaces?)
And 

```java
public interface TaskSupplier extends TaskModifier {
    <V> TaskView<V> get(int timeOut, TimeUnit timeUnit) throws InterruptedException;
    <V> void result(String taskId, V result);
    void error(String taskId, Throwable error); // necessary ?
}
```

And a remodeled proper TaskManager as a client

```java
public interface TaskManager {
    <V> TaskView<V> startTask(String taskName, Map<String, Object> properties);
    boolean stopTask(String taskId);
    Map<String, Boolean> stopAllTasks(User user);
    boolean shutdownAndAwaitTermination(int timeout, TimeUnit timeUnit) throws InterruptedException;
    <V> TaskView<V> getTask(String taskId);
    List<TaskView<?>> getTasks();
    List<TaskView<?>> getTasks(User user, Pattern pattern);
    List<TaskView<?>> clearDoneTasks();
}
```

That provides tasks to loops (consumers). We will use those interfaces as "[domain layer](https://www.martinfowler.com/bliki/PresentationDomainDataLayering.html)" that we are going to insert above the raw `Queues` that can be seen as "data layer".

Locally for example the `MemoryTaskManager` could implement both `TaskManager` and `TaskSupplier` to look like what it was before refactoring:

```java
public class TaskManagerMemory implements TaskManager, TaskSupplier {
    // ...
}
```

But task executors (BatchDownload daemon and BatchSearch daemon) will only use  `TaskSupplier`. They should converge and ideally should use the same "generic" `TaskRunner`. See #1080 

It will have impacts on datashare-client. For example there was a duplicate concept of task.id and task.name that was similar, so the commit 5dd0e1a8c0732ed819b8c234c3cbe5caa38b3f07 harmonizes that. But the json representation of a task has changed and the name field is replaced by id. That is a small change but there will be other changes to come.

![amqp_task_execution drawio](https://github.com/ICIJ/datashare/assets/551723/ba48e335-5c93-436f-af9d-37c8e0209c88)

**Comment by pirhoo** [comment]
Probably a duplicate of #1080

**Comment by bamthomas** [comment]
blocked by https://github.com/ICIJ/datashare/issues/1286
